src/completers/git.zsh

```diff
@@ -342,15 +404,22 @@ _fzf_complete_git-commits() {
     local fzf_options=$1
     shift
 
-    _fzf_complete --ansi --tiebreak=index ${(Q)${(Z+n+)fzf_options}} -- $@$prefix_option < <({
+    _fzf_complete --ansi --tiebreak=index ${(Q)${(Z+n+)fzf_options}} -- $@$prefix_option$prefix_ref < <({
         git for-each-ref refs/heads refs/remotes --format='%(refname:short) branch %(contents:subject)' 2> /dev/null
         git for-each-ref refs/tags --format='%(refname:short) tag %(contents:subject)' --sort=-version:refname 2> /dev/null
         git log --format='%h commit %s' 2> /dev/null
     } | _fzf_complete_tabularize ${fg[yellow]} ${fg[green]})
 }
 
 _fzf_complete_git-commits_post() {
-    awk '{ print $1 }'
+    local input=$(awk '{ print $1 }')
+
+    if (( $no_spacing )); then
```

Comment:
Maybe should it be using `prefix_ref` rather than sticking to a new variable `no_spacing`?

Reply:
OK, this cannot be done without a new variable. Just rename this to more semantic one such as `as_refname_prefix` and use `[[` rather than arithmetic operation.